### PR TITLE
Unit TOC

### PR DESCRIPTION
**Pull request by brittany-johnson**
Need for https://github.com/openstax/cnx-recipes/issues/3506 and all other books w/ units

### CODE REVIEW

File: lib/kitchen/directions/bake_chapter_summary.rb [hunk 1]
```diff
@@ -10,7 +10,8 @@ def self.v1(chapter:, metadata_source:)
         # TODO: include specific page types somehow without writing it out
         chapter.pages('$:not(.introduction)').each do |page|
           summary = page.summary
-          summary.first("[data-type='title']").trash # get rid of old title
+          summary.first("[data-type='title']").trash if
+            summary.first("[data-type='title']").respond_to? :trash # get rid of old title if exists
```
Comment (brittany-johnson) [inline lib/kitchen/directions/bake_chapter_summary.rb:14]: in am gov 2e, not all summaries have a title

This is not related to unit toc so if you prefer I can move out to a different PR 

Also wondering if this is the clearest way to write this?

**Comment by brittany-johnson** [comment]
The codecov report isn't displaying properly on this PR but I saw locally that this is the line that I'm missing. I'm not sure how to test this though: 
Screen Shot 2021-03-10 at 4 36 46 PM

File: lib/kitchen/directions/bake_chapter_summary.rb (hunk 1 above)
Comment (jpslav) [reply lib/kitchen/directions/bake_chapter_summary.rb:17]:
```ruby
summary.first("[data-type='title']")&.trash
```

The `&` is the [safe navigation operator](https://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) in Ruby.  It'll only call the next method if the part to the left of it is not `nil`.

Fine to have a random change here I think.

File: lib/kitchen/unit_element.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    # @param document [Document] this element's document
+    #
+    def initialize(node:, document: nil)
+      super(node: node,
+            document: document,
+            enumerator_class: UnitElementEnumerator,
+            short_type: :unit)
+    end
+
+    def title
```
Comment (jpslav) [inline lib/kitchen/unit_element.rb:20]: Add method documentation here (the YARD docs like above and below)

File: spec/directions/bake_toc_spec.rb [hunk 3]
```diff
@@ -19,51 +20,86 @@
             <span data-type="" itemprop="" class="os-text">Preface</span>
           </h1>
         </div>
-        <div data-type="chapter">
```
Comment (jpslav) [inline spec/directions/bake_toc_spec.rb:22]: Can you leave the unit-less `book1` and its spec in and then add another spec that shows that units also work?  That way we can test both; as is the test no longer tests the case without units.

**Review by jpslav — CHANGES_REQUESTED** [review]

File: spec/directions/bake_toc_spec.rb (hunk 3 above)
Comment (brittany-johnson) [reply spec/directions/bake_toc_spec.rb:22]: oop and this fixed the code coverage issue I was having 😆

File: lib/kitchen/unit_element.rb (hunk 2 above)
Comment (jpslav) [reply lib/kitchen/unit_element.rb:24]: Needs 

```
# @return [Element]
```

Comment (brittany-johnson) [reply lib/kitchen/unit_element.rb:26]: 🙈

**Review by jpslav — APPROVED** [review]

**Pull request merged**